transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -99,24 +113,78 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
     * @return a Msg object in the desired output format
     */
   def transformMetadata(entityType: String, fileName: String, metadata: Msg): Msg = {
-    // extract info from file name
+    val (entityId, entityVersion) = getEntityIdAndVersion(fileName: String)
+    Obj(
+      mutable.LinkedHashMap[Msg, Msg](
+        Str(s"${entityType}_id") -> Str(entityId),
+        Str("version") -> Str(entityVersion),
+        Str("content") -> Str(encode(metadata).getOrElse(""))
+      )
+    )
+  }
+
+  /**
+    *
+    * Extract the necessary info from a file of file-related metadata and put it into a form that
+    * makes it easy to pass in to the table format
+    *
+    * @param entityType the metadata entity type to prepend for the id field
+    * @param fileName the raw filename of the metadata file
+    * @param metadata the content of the metadata file in Msg format
+    * @return a Msg object in the desired output format
+    */
+  def transformFileMetadata(entityType: String, fileName: String, metadata: Msg): Msg = {
+    val (entityId, entityVersion) = getEntityIdAndVersion(fileName: String)
+    val dataFileName = metadata.read[Msg]("file_core").read[String]("file_name")
+    val (fileId, fileVersion) = getFileIdAndVersion(dataFileName)
+    // put values in the form we want
+    Obj(
+      mutable.LinkedHashMap[Msg, Msg](
+        Str(s"${entityType}_id") -> Str(entityId),
+        Str("version") -> Str(entityVersion),
+        Str("content") -> Str(encode(metadata).getOrElse("")),
+        Str("source_file_id") -> Str(fileId),
+        Str("source_file_version") -> Str(fileVersion)
```

Comment:
We aren't extracting the checksum field for now, right? The [importer design doc](https://docs.google.com/document/d/1AJfpaGMauFlusgAOKQHNeCta-PrHz5OObO-FcHyn_zw/edit?usp=sharing) seemed to suggest that we would hold off on using it for now, and the file metadata I looked through just had the checksum field set to null.

I wanted to confirm because the description on the [jira ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1077) suggested that we _would_ want to use some kind of content hash.

Reply:
I think we will need a checksum in both these outputs and the bulk-file request outputs. They won't have a column in BQ, but they're needed for a BQ-based diffing pre-step.

Are the checksums still null after UCSC refreshed their test bucket yesterday? If so, we should ask them about it.

Reply:
Yeah, the checksum field is still null. Before the refresh, it wasn't a part of the json file at all, now the field is listed but set to: `"checksum": null`.

I'll update my changes to use it though 👍